modules/structs/repo_compare.go

```diff
@@ -0,0 +1,68 @@
+// Copyright 2019 The Gitea Authors. All rights reserved.
+// Use of this source code is governed by a MIT-style
+// license that can be found in the LICENSE file.
+
+package structs
+
+// DiffLineSectionInfo represents diff line section meta data
+type DiffLineSectionInfo struct {
+	Path          string `json:"path"`
+	LastLeftIdx   int    `json:"last_left_idx"`
+	LastRightIdx  int    `json:"last_right_idx"`
+	LeftIdx       int    `json:"left_idx"`
+	RightIdx      int    `json:"right_idx"`
+	LeftHunkSize  int    `json:"left_hunk_size"`
+	RightHunkSize int    `json:"right_hunk_size"`
+}
+
+// DiffLine represents a line difference in a DiffSection.
+type DiffLine struct {
+	LeftIdx     int                  `json:"left_idx"`
+	RightIdx    int                  `json:"right_idx"`
+	Type        int                  `json:"type"`
+	Content     string               `json:"content"`
+	Comments    []*Comment           `json:"comments"`
+	SectionInfo *DiffLineSectionInfo `json:"section_info"`
+}
+
+// DiffSection represents a section of a DiffFile.
+type DiffSection struct {
+	Name  string      `json:"name"`
+	Lines []*DiffLine `json:"lines"`
+}
+
+// DiffFile represents a file diff.
+type DiffFile struct {
```

Comment:
pleace make it the same type as github does -> I will need this in #10933 and #10921 too ...
If you like to add extra infos github does not profide this is fine but with two limitations: the new fileds should not colide with github onec and info should not be profided elswhere (only if this would cause to mouch new api calls ...)